bin/llvm-krun

```diff
@@ -167,10 +167,12 @@ do
       debug="${debug/--args /} -x ${debug_command} --args "
     fi
     shift;
+    shift
     ;;
 
     --debug-batch)
     debug="gdb --batch --args "
```

Comment:
Prefer arrays to strings with spaces for command / argument lists, generally

Reply:
Here, we expect the path of the file, which contains the command we want to execute. I don't think we expect spaces on it.

Reply:
To elaborate a bit here @Robertorosmaninho, the issue is that building a command line as a string with spaces in it means that you need an unquoted expansion (`$debug`) later in the program. Shellcheck normally doesn't like this (as it can lead to unexpected behaviour re. globbing), but in some cases it will attempt to do some rough [data-flow analysis](https://www.shellcheck.net/wiki/SC2248) to prove that such an expansion is actually safe; your `replace_gdb_by_lldb` function has slightly vexed Shellcheck here I think. I may open a PR later on to tighten up our settings.

See https://www.shellcheck.net/wiki/SC2086 (under "Exceptions") for the full rationale here.